sw/device/silicon_creator/mask_rom/mask_rom_epmp.c

```diff
@@ -29,12 +29,27 @@ void mask_rom_epmp_state_init(epmp_state_t *state) {
       .end = TOP_EARLGREY_EFLASH_BASE_ADDR + TOP_EARLGREY_EFLASH_SIZE_BYTES};
   // TODO(#7117): generate MMIO addresses.
   const epmp_region_t mmio = {.start = 0x40000000, .end = 0x50000000};
+  const epmp_region_t debug_rom = {.start = TOP_EARLGREY_RV_DM_ROM_BASE_ADDR,
+                                   .end = TOP_EARLGREY_RV_DM_ROM_BASE_ADDR +
+                                          TOP_EARLGREY_RV_DM_ROM_SIZE_BYTES};
   const epmp_region_t stack_guard = {.start = (uintptr_t)_stack_start,
                                      .end = (uintptr_t)_stack_start + 4};
   const epmp_region_t ram = {.start = TOP_EARLGREY_RAM_MAIN_BASE_ADDR,
                              .end = TOP_EARLGREY_RAM_MAIN_BASE_ADDR +
                                     TOP_EARLGREY_RAM_MAIN_SIZE_BYTES};
 
+  epmp_perm_t debug_rom_access;
+  switch (lc_state) {
+    case kLcStateTest:
+    case kLcStateDev:
+    case kLcStateRma:
+      debug_rom_access = kEpmpPermLockedReadWriteExecute;
+      break;
+    default:
+      debug_rom_access = kEpmpPermLockedNoAccess;
+      break;
+  }
```

Comment:
We recently hardened all switch statements that use `lc_state` (disclaimer: I didn't get the chance to compile this change):

```suggestion
  epmp_perm_t debug_rom_access = kEpmpPermLockedNoAccess;
  switch (launder32(lc_state)) {
    case kLcStateTest:
      HARDENED_CHECK_EQ(lc_state, kLcStateTest);
      debug_rom_access = kEpmpPermLockedReadWriteExecute;
      break;
    case kLcStateDev:
      HARDENED_CHECK_EQ(lc_state, kLcStateDev);
      debug_rom_access = kEpmpPermLockedReadWriteExecute;
      break;
    case kLcStateProd:   
      HARDENED_CHECK_EQ(lc_state, kLcStateProd);
      debug_rom_access = kEpmpPermLockedNoAccess;
      break;
    case kLcStateProdEnd:   
      HARDENED_CHECK_EQ(lc_state, kLcStateProdEnd);
      debug_rom_access = kEpmpPermLockedNoAccess;
      break;
    case kLcStateRma:
      HARDENED_CHECK_EQ(lc_state, kLcStateRma);
      debug_rom_access = kEpmpPermLockedReadWriteExecute;
      break;
    default:
      HARDENED_UNREACHABLE();
  }
```

If you run into build issues that will take time to resolve, can you create an issue to track this?

Reply:
Updated